app/src/org/commcare/services/CommCareFirebaseMessagingService.java

```diff
@@ -189,15 +198,19 @@ private void showNotification(Map<String, String> payloadData) {
             PendingIntent contentIntent = PendingIntent.getActivity(this, 0, intent, flags);
 
             NotificationCompat.Builder fcmNotification = new NotificationCompat.Builder(this,
-                    CommCareNoficationManager.NOTIFICATION_CHANNEL_PUSH_NOTIFICATIONS_ID)
+                    notificationChannel)
                     .setContentTitle(notificationTitle)
                     .setContentText(notificationText)
                     .setContentIntent(contentIntent)
                     .setAutoCancel(true)
                     .setSmallIcon(R.drawable.commcare_actionbar_logo)
-                    .setPriority(NotificationCompat.PRIORITY_HIGH)
+                    .setPriority(priority)
                     .setWhen(System.currentTimeMillis());
 
+            if(largeIcon != null) {
+                fcmNotification.setLargeIcon(largeIcon);
```

Comment:
`setLargeIcon` accepts`null` params so should be fine to add this without a null check in the original builder chain itself.